pyonfx/shape.py

```diff
@@ -184,7 +185,7 @@ def map(
             while i < n:
                 try:
                     # Applying transformation
-                    x, y = fun(float(cmds_and_points[i]), float(cmds_and_points[i + 1]))
+                    x, y = fun(float(cmds_and_points[i]), float(cmds_and_points[i + 1]), None)
```

Comment:
I'm not able to test it with the library, but this should return the error:
`TypeError: <lambda>() takes 2 positional arguments but 3 were given`

We're in the branch where `fun` only accepted 2 parameters and you're passing to it 3.

Reply:
According to the type of the method it's `fun: Callable[[float, float, Optional[str]]` so it needs 3 arguments.

This third argument is also overrided at L206. I'm not sure if it's intented.

Reply:
In this case the type was simply wrong. The correct type would have been
```
fun: Union[Callable[[float, float], Tuple[float, float]], Callable[[float, float, str], Tuple[float, float]]]
```
which would have allowed the function to take either 2 or 3 arguments. That is also the reason, why you had to add all these `_` in https://github.com/Varde-s-Forks/PyonFX/commit/8266078a7690db476d23cebf3128e22ac96f79e1.

Reply:
Totally agree with @Funami580.